src/lib/find-vnodes-by-ref.js

```diff
@@ -0,0 +1,15 @@
+// @flow
+
+import { removeDuplicateNodes, findAllVNodes } from './vnode-utils'
+
+function nodeMatchesRef (node: VNode, refName: string): boolean {
+  return node.data && node.data.ref === refName
+}
+
+export default function findVNodesByRef (vNode: VNode, refName: string): Array<VNode> {
+  const nodes = findAllVNodes(vNode)
+  const refFilteredNodes = nodes.filter(node => nodeMatchesRef(node, refName))
+  // Only return ref matches from top level vNode
```

Comment:
For my own understanding, why do we not do a recursive check here? Might be helpful to document in the comment.

Reply:
Updated. This is because we want to emulate the exact selector functionality of `wrappedComponent.$refs.myRef`.

If we do a recursive check, we may return a general result of `wrappedComponent.$refs.someComponent.$refs.someComponent.$refs.myRef` when we only want to grab ref elements on the top-level wrapper itself. If you need a sub-component ref, you'd need to find the subcomponent and get the ref off of it explicitly.

Reply:
Cool, thanks for the 💯 explanation!